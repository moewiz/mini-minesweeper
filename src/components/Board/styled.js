import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

export const BoardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-top: 30px;
`;

export const LoadingStyled = styled.span`
  font-size: 20px;
`;

export const ErrorStyled = styled.span`
  font-size: 20px;
  color: #ff4141;
`;

export const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 30px;
`;

const HeaderBtn = css`
  color: #000;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;

export const NewGameStyled = styled.div`
  ${HeaderBtn}
`;

export const HomeStyled = styled(Link)`
  ${HeaderBtn}
`;
