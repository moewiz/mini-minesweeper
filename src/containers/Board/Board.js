import React, { useEffect } from "react";
import PropTypes from "prop-types";
import { selectors as MiniMineSweeperActionsSelectors } from "../../reducers/miniMineSweeper";
import { useShallowEqualSelector } from "../../utils/hookRecipes";
import { Board } from "../../components";

const BoardContainer = ({ gameLevel, setupGame, ...props }) => {
  useEffect(() => {
    setupGame();
  }, [setupGame]);

  const matrices = useShallowEqualSelector(
    MiniMineSweeperActionsSelectors.getMatrices
  );
  const loading = useShallowEqualSelector(
    MiniMineSweeperActionsSelectors.getLoading
  );
  const error = useShallowEqualSelector(
    MiniMineSweeperActionsSelectors.getError
  );

  return (
    <Board
      {...props}
      onNewGame={setupGame}
      gameLevel={gameLevel}
      matrices={matrices}
      loading={loading}
      error={error}
    />
  );
};

BoardContainer.propTypes = {
  gameLevel: PropTypes.shape({
    size: PropTypes.number,
    mines: PropTypes.number
  }).isRequired
};

export default BoardContainer;
