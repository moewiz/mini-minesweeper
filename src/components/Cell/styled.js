import styled, { css } from "styled-components";

export const CellWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  border: 1px solid;
  border-color: #000;
  background: ${props => (props.isOpen ? "#e4e4e4" : "#fff")};
  font-weight: 600;
  ${props =>
    props.isMine &&
    props.isOpen &&
    css`
      position: relative;
      &:after {
        content: "";
        position: absolute;
        background: #24292e !important;
        border: 1px solid #24292e;
        border-radius: 50%;
        top: 4px;
        left: 4px;
        width: 20px;
        height: 20px;
      }
    `}
  &:hover {
    background: #e4e4e4;
  }
`;
