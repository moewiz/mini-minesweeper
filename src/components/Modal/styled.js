import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

export const ButtonWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const HeaderBtn = css`
  color: #000;
  cursor: pointer;
  font-size: 20px;
  padding: 10px;
  text-decoration: none;
  &:hover {
    text-decoration: underline;
  }
`;

export const NewGameStyled = styled.div`
  ${HeaderBtn}
`;

export const HomeLinkStyled = styled(Link)`
  ${HeaderBtn}
`;

export const MessageStyled = styled.div`
  font-size: 20px;
  text-align: center;
  margin-top: 5px;
`;
