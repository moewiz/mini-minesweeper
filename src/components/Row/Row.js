import React from "react";
import PropTypes from "prop-types";
import { Cell } from "../../containers";
import { RowWrapper } from "./styled";

const Row = ({ row }) => {
  return (
    <RowWrapper>
      {row.map((cell, index) => (
        <Cell cell={cell} key={index} />
      ))}
    </RowWrapper>
  );
};

Row.propTypes = {
  row: PropTypes.arrayOf(
    PropTypes.shape({
      minesAround: PropTypes.number,
      isOpen: PropTypes.bool.isRequired,
      x: PropTypes.number.isRequired,
      y: PropTypes.number.isRequired
    })
  ).isRequired
};

export default Row;
