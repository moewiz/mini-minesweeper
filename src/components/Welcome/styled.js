import styled from "styled-components";
import { Link } from "react-router-dom";

export const HomeWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0 auto;
  margin-top: 30px;
`;

export const GroupLevelStyled = styled.div`
  display: flex;
  flex-direction: row;
  width: 350px;
  justify-content: space-around;
`;

export const LevelBtnStyled = styled(Link)`
  font-size: 20px;
  color: #000;
  border: 1px solid;
  padding: 15px;
  text-decoration: none;
  display: flex;
  &:hover {
    text-decoration: underline;
  }
`;
