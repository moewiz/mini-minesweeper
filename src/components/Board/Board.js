import React from "react";
import PropTypes from "prop-types";
import Row from "../Row";
import { BoardWrapper, LoadingStyled, ErrorStyled } from "./styled";

const Board = ({ matrices, gameLevel, loading, error }) => {
  const width = gameLevel.size * 32 + 50;

  if (loading) {
    return (
      <BoardWrapper>
        <LoadingStyled>Setting up the game</LoadingStyled>
      </BoardWrapper>
    );
  }

  if (error) {
    return (
      <BoardWrapper>
        <ErrorStyled>{error}</ErrorStyled>
      </BoardWrapper>
    );
  }

  return (
    <BoardWrapper style={{ width: `${width}px` }}>
      {matrices.map((row, index) => (
        <Row row={row} key={index} />
      ))}
    </BoardWrapper>
  );
};

Board.propTypes = {
  matrices: PropTypes.arrayOf(
    PropTypes.arrayOf(
      PropTypes.shape({
        x: PropTypes.number.isRequired,
        y: PropTypes.number.isRequired,
        minesAround: PropTypes.number.isRequired,
        isOpen: PropTypes.bool.isRequired
      })
    )
  ).isRequired,
  gameLevel: PropTypes.shape({
    size: PropTypes.number,
    mines: PropTypes.number
  }).isRequired,
  loading: PropTypes.bool.isRequired,
  error: PropTypes.string
};

export default Board;
