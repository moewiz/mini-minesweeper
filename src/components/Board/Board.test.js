import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import Board from "./Board";
import Row from "../Row";
import { NewGameStyled } from "./styled";

describe("<Board />", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  const onNewGameMock = jest.fn();
  const Component = mount(
    <Router>
      <Board
        onNewGame={onNewGameMock}
        matrices={[[], [], [], []]}
        loading={false}
        error={null}
      />
    </Router>
  );

  it("should render without throwing an error", () => {
    expect(Component.length).toBe(1);
  });

  it("should match snapshot", () => {
    expect(Component).toMatchSnapshot();
  });

  it("should call onNewGameMock 1 time when New game button is clicked", () => {
    const NewGameBtn = Component.find(NewGameStyled).at(0);
    NewGameBtn.simulate("click");
    expect(onNewGameMock).toBeCalledTimes(1);
  });

  it("should render 4 Row component", () => {
    const Rows = Component.find(Row);
    expect(Rows.length).toBe(4);
  });
});
