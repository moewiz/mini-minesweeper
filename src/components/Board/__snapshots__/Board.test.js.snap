// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Board
      error={null}
      loading={false}
      matrices={
        Array [
          Array [],
          Array [],
          Array [],
          Array [],
        ]
      }
      onNewGame={[MockFunction]}
    >
      <styled.div>
        <div
          className="sc-fzplWN gYxPsW"
        >
          <styled.div
            onClick={[MockFunction]}
          >
            <div
              className="sc-fznyAO OjnoA"
              onClick={[MockFunction]}
            >
              New game
            </div>
          </styled.div>
          <Styled(Link)
            to="/"
          >
            <Link
              className="sc-fznKkj eKGikx"
              to="/"
            >
              <LinkAnchor
                className="sc-fznKkj eKGikx"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="sc-fznKkj eKGikx"
                  href="/"
                  onClick={[Function]}
                >
                  Home page
                </a>
              </LinkAnchor>
            </Link>
          </Styled(Link)>
        </div>
      </styled.div>
      <styled.div>
        <div
          className="sc-fzozJi cArkNM"
        >
          <Row
            key="0"
            row={Array []}
          >
            <styled.div>
              <div
                className="sc-AxmLO MKJBh"
              />
            </styled.div>
          </Row>
          <Row
            key="1"
            row={Array []}
          >
            <styled.div>
              <div
                className="sc-AxmLO MKJBh"
              />
            </styled.div>
          </Row>
          <Row
            key="2"
            row={Array []}
          >
            <styled.div>
              <div
                className="sc-AxmLO MKJBh"
              />
            </styled.div>
          </Row>
          <Row
            key="3"
            row={Array []}
          >
            <styled.div>
              <div
                className="sc-AxmLO MKJBh"
              />
            </styled.div>
          </Row>
        </div>
      </styled.div>
    </Board>
  </Router>
</BrowserRouter>
`;
